TemplateApplication/Schedule/ScheduleView.swift

```diff
@@ -10,54 +10,53 @@ import OrderedCollections
 @_spi(TestingSupport) import SpeziAccount
 import SpeziQuestionnaire
 import SpeziScheduler
+import SpeziSchedulerUI
 import SwiftUI
 
 
-struct ScheduleView: View {
+struct EventView: View {
+    private let event: Event
+
     @Environment(TemplateApplicationStandard.self) private var standard
-    @Environment(TemplateApplicationScheduler.self) private var scheduler
-    @Environment(Account.self) private var account: Account?
+    @Environment(\.dismiss) private var dismiss
 
-    @State private var presentedContext: EventContext?
-    @Binding private var presentingAccount: Bool
-    
-    
-    @MainActor private var eventContextsByDate: OrderedDictionary<Date, [EventContext]> {
-        let eventContexts = scheduler.tasks.flatMap { task in
-            task
-                .events(
-                    from: Calendar.current.startOfDay(for: .now),
-                    to: .numberOfEventsOrEndDate(100, .now)
-                )
-                .map { event in
-                    EventContext(event: event, task: task)
+    var body: some View {
+        if let questionnaire = event.task.questionnaire {
+            QuestionnaireView(questionnaire: questionnaire) { result in
+                dismiss()
+
+                guard case let .completed(response) = result else {
+                    return // user cancelled the task
                 }
-        }
-            .sorted()
 
-        return OrderedDictionary(grouping: eventContexts) { eventContext in
-            Calendar.current.startOfDay(for: eventContext.event.scheduledAt)
+                event.complete()
+                await standard.add(response: response)
+            }
         }
     }
 
+    init(_ event: Event) {
+        self.event = event
+    }
+}
+
+
+struct ScheduleView: View {
+    @Environment(Account.self) private var account: Account?
+
+    @State private var presentedEvent: Event?
+    @Binding private var presentingAccount: Bool
+
     
     var body: some View {
         NavigationStack {
-            let eventContextsByDate = eventContextsByDate
-            List(eventContextsByDate.keys, id: \.timeIntervalSinceNow) { startOfDay in
-                Section(format(startOfDay: startOfDay)) {
-                    ForEach(eventContextsByDate[startOfDay] ?? [], id: \.event) { eventContext in
-                        EventContextView(eventContext: eventContext)
-                            .onTapGesture {
-                                if !eventContext.event.complete {
-                                    presentedContext = eventContext
-                                }
-                            }
-                    }
+            TodayList { event in
+                InstructionsTile(event) { // TODO: use TASK_CONTEXT_ACTION_QUESTIONNAIRE for the footer Action button!
```

Comment:
Very cool to see the simplification in here!

Pointing out the TODO remaining here.

Reply:
Resolved 🚀